src/System.Management.Automation/engine/regex.cs

```diff
@@ -278,6 +278,64 @@ public static bool ContainsWildcardCharacters(string pattern)
             return result;
         }
 
+        /// FIXME
+        /// For invalid wildcard pattern, it currently only check if there is
+        /// unclosed bracket. Need to find a better solution and rewrite this.
+        ///
+        /// <summary>
+        /// Checks to see if the given string might contains wildcard pattern.
+        /// </summary>
+        /// <param name="pattern">
+        /// String which needs to be checked
+        /// </param>
+        /// <returns>
+        /// true if the string does not contain invalid wildcard pattern,
+        /// false otherwise.
+        /// </returns>
+        /// <remarks>
+        /// Currently string contains { '*', '?' } or both { '[', ']' } is
+        /// considered to be properbly a wildcard pattern and
+        /// '`' is the escape character.
+        /// </remarks>
+        public static bool ContainsValidWildcardPattern(string pattern)
```

Comment:
Before finalizing the behavior of this method, let's make it internal.

Reply:
Yes.